doc: add type stripping to ambassadors program (#56598)
nodejs/node: doc/contributing/advocacy-ambassador-program.md

* @marco-ippolito

You can find their contact email in the [`README.md`](../../README.md#tsc-technical-steering-committee)

#### Node.js Type Stripping

##### Goal

The goal is to raise awareness of the Node.js TypeScript Type Stripping in the JavaScript ecosystem.
Some of the things to highlight include:

> * The new feature and its benefits and limitations.

Reviewer: nit: I find it weird to ask to highlight the whole feature, it's kind of already covered by the previous sentence
```suggestion
* The benefits and limitations of the current implementation.
```

> * The tsconfig.json configuration options to use in combination for type checking.
> * New relevant TypeScript compiler options.

Reviewer: Aren't those the same?
```suggestion
* The `tsconfig.json` configuration options to use in combination for type checking.
```
